packages/integration/src/azure/config.ts

```diff
@@ -117,45 +199,117 @@ export function readAzureIntegrationConfig(
   config: Config,
 ): AzureIntegrationConfig {
   const host = config.getOptionalString('host') ?? AZURE_HOST;
+
+  let credentials = config
+    .getOptionalConfigArray('credentials')
+    ?.map(credential => {
+      const result: Partial<AzureDevOpsCredential> = {
+        organizations: credential.getOptionalStringArray('organizations'),
+        token: credential.getOptionalString('token'),
+        tenantId: credential.getOptionalString('tenantId'),
+        clientId: credential.getOptionalString('clientId'),
+        clientSecret: credential.getOptionalString('clientSecret'),
+      };
+
+      return result;
+    });
+
   const token = config.getOptionalString('token');
 
-  const credential = config.getOptional<AzureCredential>('credential')
-    ? {
+  if (token !== undefined) {
+    const mapped = [{ token: token }];
+    credentials = credentials?.concat(mapped) ?? mapped;
+  }
+
+  if (config.getOptional('credential') !== undefined) {
+    const mapped = [
+      {
+        organizations: config.getOptionalStringArray(
+          'credential.organizations',
+        ),
+        token: config.getOptionalString('credential.token'),
         tenantId: config.getOptionalString('credential.tenantId'),
         clientId: config.getOptionalString('credential.clientId'),
         clientSecret: config.getOptionalString('credential.clientSecret'),
-      }
-    : undefined;
+      },
+    ];
+    credentials = credentials?.concat(mapped) ?? mapped;
```

Comment:
You are right; the `credential` field should go away. The only reason I did not remove it already is to have a deprecation period instead of a breaking change. A credentials with no `organizations` will be used as a fallback credential so it makes no sense to support both `credential` and `credentials`